listFailedObjects.js

```diff
@@ -32,31 +33,31 @@ if (!SECRET_KEY) {
     process.exit(1);
 }
 
-const s3 = new AWS.S3({
-    accessKeyId: ACCESS_KEY,
-    secretAccessKey: SECRET_KEY,
+const s3 = new S3Client({
     region: 'us-east-1',
-    sslEnabled: false,
-    endpoint: ENDPOINT,
-    s3ForcePathStyle: true,
-    apiVersions: { s3: '2006-03-01' },
-    signatureVersion: 'v4',
-    signatureCache: false,
-    httpOptions: {
-        maxRetries: 0,
-        timeout: 0,
-        agent: new http.Agent({ keepAlive: true }),
+    credentials: {
+        accessKeyId: ACCESS_KEY,
+        secretAccessKey: SECRET_KEY,
     },
+    endpoint: ENDPOINT,
+    forcePathStyle: true,
+    tls: false,
+    requestHandler: new NodeHttpHandler({
+        httpAgent: new http.Agent({ keepAlive: true }),
+        requestTimeout: 60000,
+    }),
 });
 
 // list object versions
 function _listObjectVersions(bucket, VersionIdMarker, KeyMarker, cb) {
-    s3.listObjectVersions({
+    s3.send(new ListObjectVersionsCommand({
         Bucket: bucket,
         MaxKeys: LISTING_LIMIT,
         VersionIdMarker,
         KeyMarker,
-    }, cb);
+    }))
+        .then(data => cb(null, data))
+        .catch(err => cb(err));
```

Comment:
nit: maybe just me, but I think the "gap" very confusing to read, breaking the flow of the chained call...
(which there was a strict linter, to enforce such "style" decision)

```suggestion
    })).then(data => cb(null, data)).catch(cb);
```